### PR TITLE
 fix parse errors found by fuzzing test

### PR DESCRIPTION
**Pull request by lafirest**
Fixes <issue-number>

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/emqx/emqx/blob/master/CONTRIBUTING.md.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Added tests for the changes
- [x] Changed lines covered in coverage report
- [x] Change log has been added to `changes/` dir
- [x] ~~For EMQX 4.x: `appup` files updated (execute `scripts/update-appup.sh emqx`)~~
- [x] For internal contributor: there is a jira ticket to track this change
      [EMQX-8305](https://emqx.atlassian.net/browse/EMQX-8305)
- [x] ~~If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up~~
- [x] ~~In case of non-backward compatible changes, reviewer should check this item as a write-off, and add details in **Backward Compatibility** section~~

### CODE REVIEW

**Review by savonarola — DISMISSED** [review]

File: changes/v5.0.12-en.md [hunk 1]
```diff
@@ -14,3 +14,5 @@
 ## Bug fixes
 
 - Fix that the obsolete SSL files aren't deleted after the ExHook config update [#9432](https://github.com/emqx/emqx/pull/9432).
+
+- Fix some potential parse errors [#9444](https://github.com/emqx/emqx/pull/9444).
```
Comment (zmstone) [inline changes/v5.0.12-en.md:18]:
```suggestion
- Fix some potential MQTT packet parse errors [#9444](https://github.com/emqx/emqx/pull/9444).
```

File: apps/emqx/src/emqx_frame.erl [hunk 2]
```diff
@@ -411,7 +432,9 @@ parse_packet(
     #{strict_mode := StrictMode, version := ?MQTT_PROTO_V5}
 ) ->
     {Properties, <<>>} = parse_properties(Rest, ?MQTT_PROTO_V5, StrictMode),
-    #mqtt_packet_auth{reason_code = ReasonCode, properties = Properties}.
+    #mqtt_packet_auth{reason_code = ReasonCode, properties = Properties};
+parse_packet(_Header, _FrameBin, _Options) ->
+    ?PARSE_ERR(incorrect_packet).
```
Comment (zmstone) [inline apps/emqx/src/emqx_frame.erl:437]:
```suggestion
    ?PARSE_ERR(malformed_packet).
```

File: apps/emqx/test/emqx_ws_connection_SUITE.erl [hunk 3]
```diff
@@ -535,7 +535,7 @@ t_parse_incoming(_) ->
 
 t_parse_incoming_frame_error(_) ->
     {Packets, _St} = ?ws_conn:parse_incoming(<<3, 2, 1, 0>>, [], st()),
-    FrameError = {frame_error, function_clause},
+    FrameError = {frame_error, incorrect_packet},
```
Comment (zmstone) [inline apps/emqx/test/emqx_ws_connection_SUITE.erl:538]:
```suggestion
    FrameError = {frame_error, malformed_packet},
```

File: apps/emqx/src/emqx_frame.erl [hunk 4]
```diff
@@ -260,42 +271,52 @@ parse_packet(
...
-    };
+    case parse_properties(Rest, Ver, StrictMode) of
+        {Properties, <<>>} ->
+            #mqtt_packet_connack{
+                ack_flags = AckFlags,
+                reason_code = ReasonCode,
+                properties = Properties
+            };
+        _ ->
+            ?PARSE_ERR(unexpected_properties_data)
```
Comment (zmstone) [inline apps/emqx/src/emqx_frame.erl:318]: this clause is lack of test coverage ?
at least no test case covering this error reason.
```suggestion
            ?PARSE_ERR(malformed_properties)
```

File: apps/emqx/src/emqx_frame.erl [hunk 5]
```diff
@@ -260,42 +271,52 @@ parse_packet(
...
-        will_retain = bool(WillRetain),
-        keepalive = KeepAlive,
-        properties = Properties,
-        clientid = ClientId
-    },
-    {ConnPacket1, Rest5} = parse_will_message(ConnPacket, Rest4, StrictMode),
-    {Username, Rest6} = parse_utf8_string(Rest5, StrictMode, bool(UsernameFlag)),
-    {Password, <<>>} = parse_utf8_string(Rest6, StrictMode, bool(PasswordFlag)),
-    ConnPacket1#mqtt_packet_connect{username = Username, password = Password};
+    case parse_utf8_string(FrameBin, StrictMode) of
```
Comment (zmstone) [inline apps/emqx/src/emqx_frame.erl:274]:
```suggestion
    {ProtoName, Rest} = parse_utf8_string(FrameBin, StrictMode),
    case Rest of
        ...
```

File: apps/emqx/src/emqx_frame.erl [hunk 6]
```diff
@@ -260,42 +271,52 @@ parse_packet(
...
+                properties = Properties,
+                clientid = ClientId
+            },
+            {ConnPacket1, Rest5} = parse_will_message(ConnPacket, Rest4, StrictMode),
+            {Username, Rest6} = parse_utf8_string(Rest5, StrictMode, bool(UsernameFlag)),
+            case parse_utf8_string(Rest6, StrictMode, bool(PasswordFlag)) of
+                {Password, <<>>} ->
+                    ConnPacket1#mqtt_packet_connect{username = Username, password = Password};
+                _ ->
+                    ?PARSE_ERR(unexpected_data)
```
Comment (zmstone) [inline apps/emqx/src/emqx_frame.erl:300]:
```suggestion
                    ?PARSE_ERR(malformed_connect_payload)
```

File: apps/emqx/src/emqx_frame.erl [hunk 7]
```diff
@@ -260,42 +271,52 @@ parse_packet(
...
+            end;
+        _ ->
+            ?PARSE_ERR(incorrect_data)
```
Comment (zmstone) [inline apps/emqx/src/emqx_frame.erl:303]:
```suggestion
            ?PARSE_ERR(malformed_connect_header)
```

File: apps/emqx/src/emqx_frame.erl (hunk 5 above)
Comment (zmstone) [inline apps/emqx/src/emqx_frame.erl:274]: refactor this clause to a smaller function.

**Review by zmstone — CHANGES_REQUESTED** [review]

File: apps/emqx/src/emqx_frame.erl (hunk 5 above)
Comment (lafirest) [reply apps/emqx/src/emqx_frame.erl]: I think there is small enough, I don't know how to simplify it.

**Pull request closed without merging**